EIPS/eip-4788.md

```diff
@@ -77,76 +79,165 @@ When verifying a block, execution clients **MUST** ensure the root value in the
 
 For a genesis block with no existing parent beacon block root the 32 zero bytes are used as a root placeholder.
 
-### EVM changes
+#### Beacon roots contract
 
-#### Block processing
+The beacon roots contract has two operations: `get` and `set`. The input itself is not used to determine which function to execute, for that the result of `caller` is used. If `caller` is equal to `SYSTEM_ADDRESS` then the operation to perform is `set`. Otherwise, `get`.
 
-At the start of processing any execution block where `block.timestamp >= FORK_TIMESTAMP` (i.e. before processing any transactions),
-write the parent beacon root provided in the block header into the storage of the contract at `HISTORY_STORAGE_ADDRESS`.
+##### `get`
 
-In order to bound the storage used by this precompile, two ring buffers are used: one to track the latest timestamp at a given index in the ring buffer and another to track
-the latest root at a given index.
+* If `caller` is equal to `SYSTEM_ADDRESS`, the contract must revert.
```

Comment:
This seems unnecessary. The outer of `get`/`set` does the routing *based* on the `caller` so by definition the `caller` won't be `SYSTEM_ADDRESS`

(same for parallel in `set`)

Reply:
good point - updated